kcfinder/integration/civicrm.php

```diff
@@ -108,6 +111,39 @@ function authenticate_drupal($config) {
   return false;
 }
 
+/**
+ * If the user is already logged into Backdrop, bootstrap
+ * Backdrop with this user's permissions.
+ */
+function authenticate_backdrop($config) {
+  global $base_url;
+  $base_root = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') ? 'https' : 'http';
+  $base_url = $base_root .= '://'. preg_replace('/[^a-z0-9-:._]/i', '', $_SERVER['HTTP_HOST']);
+
+  if ($dir = trim(dirname($_SERVER['SCRIPT_NAME']), '\,/')) {
+    $base_path = "/$dir";
+    $base_url .= $base_path;
+  }
+
+  // Correct base_url so it points to Backdrop root.
+  $pos = strpos($base_url, '/sites/');
+  if ($pos === FALSE) {
+    $pos = strpos($base_url, '/profiles/');
+  }
+  if ($pos === FALSE) {
+    $pos = strpos($base_url, '/modules/');
+  }
+  $base_url = substr($base_url, 0, $pos); // Backdrop root absolute url
+
+  CRM_Utils_System::loadBootStrap(CRM_Core_DAO::$_nullArray, true, false);
```

Comment:
looks like we can just pass an empty array without having to use the static. Note that in rare cases CRM_Core_DAO::$_nullArray can become contaminated.